h/services/annotation_metadata.py

```diff
@@ -0,0 +1,49 @@
+import os
+
+import json
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy.orm import Session
+from sqlalchemy import select, literal_column, func
+
+from h.models import Annotation, AnnotationMetadata, User, AnnotationSlim
+from h.security import decrypt_jwe_dict
+
+
+class AnnotationMetadataService:
+    def __init__(self, db: Session):
+        self._db = db
+        self._secret = os.environ.get("JWE_SECRET_LMS")
+
+    def set_annotation_metadata_from_jwe(self, annotation: Annotation, jwe: str):
+        """
+        Set the decrypted `jwe` dict as the metadata for `annotation`.
+
+        It will create a new AnnotationMetadata row or update the existing one.
+        """
+        annotation_metadata = decrypt_jwe_dict(self._secret, jwe)
+
+        self._set_annotation_metadata(annotation, annotation_metadata)
+        return annotation_metadata
+
+    def _set_annotation_metadata(self, annotation: Annotation, metadata: dict):
```

Comment:
This and the public method are based on Annotation not AnnotaionSlim even if this table is related to the later.

I reckon keeping the details of keeping `AnnotationSlim` in sync hidden is worth the extra complexity here (the JOIN on AnnotationSlim)